torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -7924,3 +7925,53 @@ def test_ddp_new_tensor_in_fwd(self):
         )
         def test_ddp_new_tensor_in_fwd_static_graph(self):
             return self._test_ddp_new_tensor_in_fwd(static_graph=True)
+
+        @sandcastle_skip_if(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "Only Nccl & Gloo backend support DistributedDataParallel",
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_stateless_api_with_ddp(self):
+            class MockModule(torch.nn.Module):
+                def __init__(self):
+                    super().__init__()
+                    self.l1 = torch.nn.Linear(1, 1)
+                    buffer = torch.ones(1)
+                    self.register_buffer('buffer', buffer)
+
+                def forward(self, x):
+                    return self.l1(x) + self.buffer
+
+            device = self.rank
+            module = MockModule().to(device)
+            module = torch.nn.parallel.DistributedDataParallel(
+                module,
+                device_ids=[device]
+            )
+            x = torch.rand((1, 1)).to(device)
+            weight = torch.nn.Parameter(torch.tensor([[1.0]])).to(device)
+            bias = torch.nn.Parameter(torch.tensor([0.0])).to(device)
+            weight.retain_grad()
```

Comment:
create the Tensors above directly on the correct device to avoid the need for this.
Also no need for them to be parameters, they can just have `requires_grad=True`:
`weight = torch.tensor([[1.]], device=device, requires_grad=True)`